tests/ashley/machina_extensions/forum/test_forum_view.py

```diff
@@ -0,0 +1,411 @@
+                        "url_remove": "?o=-2.-3",
+                        "url_toggle": "?o=-2.-0.-3",
+                        "class_attrib": "sorted ascending",
+                    },
+                    {
+                        "url_primary": "?o=1.-2.0.-3",
+                        "url_remove": "?o=-2.0.-3",
+                        "url_toggle": "?o=-2.0.-3",
+                    },
+                    {
+                        "sorted": True,
+                        "ascending": False,
+                        "sort_priority": 1,
+                        "url_primary": "?o=2.0.-3",
+                        "url_remove": "?o=0.-3",
+                        "url_toggle": "?o=2.0.-3",
+                        "class_attrib": "sorted descending",
+                    },
+                    {
+                        "sorted": True,
+                        "ascending": False,
+                        "sort_priority": 3,
+                        "url_primary": "?o=3.-2.0",
+                        "url_remove": "?o=-2.0",
+                        "url_toggle": "?o=-2.0.3",
+                        "class_attrib": "sorted descending",
+                    },
+                ],
+            ),
+        )
+
+    def test_sorting_unknown_index_column(self):
+        """ try to request an order on a column that doesn't exist """
+        # there's only 4 columns, o=3 is the maximum
+        url = f"{self.url_list_topic}?o=5"
+        # Run
+        response = self.client.get(url)
+        # Check
+        assert response.status_code == 200
+
+        # Should have 0 sorted column as the one asked is unknown
+        self.assertContains(response, "sorted ", count=0)
+
+    def test_sorting_on_label_column(self):
+        """try to request an order on a column with a label instead of a number"""
+        # there's only 4 columns, o=3 is the maximum and a number is expected
+        # point is used to separate fields
+        url = f"{self.url_list_topic}?o=DATE,POST,TOPIC"
+        # Run
+        response = self.client.get(url)
+        # Check
+        assert response.status_code == 200
+
+        # Should have 0 sorted column as the one asked is unknown
+        self.assertContains(response, "sorted ", count=0)
+
+    def test_sorting_on_mixing_type_column(self):
+        """try to request mixin good and bad params"""
+        # there's only 4 columns, o=3 is the maximum and a number is expected
+        url = f"{self.url_list_topic}?o=DATE.0.7"
+        # Run
+        response = self.client.get(url)
+        # Check
+        assert response.status_code == 200
+
+        # Should have 1 sorted column as the other two are unknown format
+        self.assertContains(response, "sorted ", count=1)
+
+        # it should be in alphabetic order
+        self.assertContentBefore(response, "TOPIC A", "TOPIC B")
+        self.assertContentBefore(response, "TOPIC B", "TOPIC C")
+
+    def test_sorting_context_wisiwig(self):
```

Comment:
I don't understand the concept of WYSIWYG in this context ^^

```suggestion
    def test_sorting_context_wysiwyg(self):
```

Reply:
that is the test I'm talking about, it links my object with the what you see in the template and validates that the links that are generated are properly shown

Reply:
no problem madmatah :-p